src/intervals/functions.jl

```diff
@@ -4,61 +4,140 @@
 # Integer power:
 function ^{T<:Real}(a::Interval{T}, n::Integer)
     isempty(a) && return a
-    n < 0  && return inv(a^(-n))
     n == 0 && return one(a)
     n == 1 && return a
-
-    iseven(n) && return @round(T, mig(a)^n, mag(a)^n)  # even power
-
-    @round(T, a.lo^n, a.hi^n)  # odd power
+    n == 2 && return sqr(a)
+    n < 0 && a == zero(a) && return emptyinterval(a)
+
+    if isodd(n) # odd power
+        isentire(a) && return a
+        if n > 0
+            a.lo == zero(T) && return @controlled_round(T, zero(T), a.hi^n)
+            a.hi == zero(T) && return @controlled_round(T, a.lo^n, zero(T))
+            return @controlled_round(T, a.lo^n, a.hi^n)
+        else
+            if a.lo ≥ zero(T)
+                a.lo == zero(T) && return @round(T, a.hi^n, convert(T,Inf))
+                return @round(T, a.hi^n, a.lo^n)
+            elseif a.hi ≤ zero(T)
+                a.hi == zero(T) && return @round(T, convert(T,-Inf), a.lo^n)
+                return @round(T, a.hi^n, a.lo^n)
+            else
+                return entireinterval(a)
+            end
+        end
+    else # even power
+        if n > 0
+            if a.lo ≥ zero(T)
+                return @controlled_round(T, a.lo^n, a.hi^n)
+            elseif a.hi ≤ zero(T)
+                return @controlled_round(T, a.hi^n, a.lo^n)
+            else
+                return @controlled_round(T, mig(a)^n, mag(a)^n)
+            end
+        else
+            if a.lo ≥ zero(T)
+                return @round(T, a.hi^n, a.lo^n)
+            elseif a.hi ≤ zero(T)
+                return @round(T, a.lo^n, a.hi^n)
+            else
+                return @round(T, mag(a)^n, mig(a)^n)
+            end
+        end
+    end
 end
-
-function ^{T<:Real}(a::Interval{T}, r::Rational)
+function sqr{T<:Real}(a::Interval{T})
     isempty(a) && return a
-    r < zero(r)  && return inv(a^(-r))
-    r == zero(r) && return one(a)
-    r == one(r)  && return a
-    isinteger(r) && return a^(round(Int,r))
+    if a.lo ≥ zero(T)
+        !isthin(a) && return @controlled_round(T, a.lo^2, a.hi^2)
+        return a*a
+    elseif a.hi ≤ zero(T)
+        !isthin(a) && return @controlled_round(T, a.hi^2, a.lo^2)
+        return a*a
+    else
+        return @controlled_round(T, mig(a)^2, mag(a)*mag(a))
+    end
+end
+
+# Floatingpoint power of a Float64/BigFloat interval:
+@compat function ^{T<:Union{Float64,BigFloat}}(a::Interval{T}, x::AbstractFloat)
```

Comment:
`T <: AbstractFloat`

By the way, maybe we should check what happens if we try to make intervals of `Float32` or `Float16`.